hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/MerkleSchemaRegistry.java

```diff
@@ -214,7 +227,8 @@ public void migrate(
             remainingStates.removeAll(statesToRemove);
             final var newStates = new FilteredWritableStates(writeableStates, remainingStates);
 
-            final var migrationContext = new MigrationContextImpl(previousStates, newStates, config);
+            final var migrationContext =
+                    new MigrationContextImpl(previousStates, newStates, config, genesisRecordsConsensusHook);
```

Comment:
Since we are using in `MigrationContextImpl`, it need not be just genesis. Should we rename this `genesisRecordsConsensusHook` to `MigrationRecordBuilderHook`

Reply:
I debated about this. The reason I decided on genesis records is because this hook is _only_ supposed to address  things created at genesis. Instead of having a generic migration record builder we would want to build different specific hooks for each specific migration, the reason being that every migration will be different and is very difficult to generalize to something useful. Thoughts?